File: Creational/Abstract Factory/Python/factory/sedan_factory.py
# factory/sedan_factory.py
from factory import IAutoBodyFactory
from product.sedan_parts import SedanLeftDoor, SedanRightDoor, SedanHood

class SedanFactory(IAutoBodyFactory):
    def create_left_door(self) -> SedanLeftDoor:
        return SedanLeftDoor()

    def create_right_door(self) -> SedanRightDoor:
        return SedanRightDoor()

    def create_hood(self) -> SedanHood:
        return SedanHood()

File: Creational/Abstract Factory/TypeScript/src/factory/SedanFactory.ts
// src/factory/SedanFactory.ts
import { IAutoBodyFactory } from './IAutoBodyFactory';
import { AutoBodyPart } from '../product/AutoBodyPart';
import { SedanLeftDoor, SedanRightDoor, SedanHood } from '../product/SedanParts';

export class SedanFactory implements IAutoBodyFactory {
    createLeftDoor(): AutoBodyPart {
        return new SedanLeftDoor();
    }

    createRightDoor(): AutoBodyPart {
        return new SedanRightDoor();
    }

    createHood(): AutoBodyPart {
        return new SedanHood();
    }
}
